ViewModel/EnvironmentAndExtendStoreUuid.php

```diff
@@ -88,35 +89,53 @@ public function getExtendStoreUuid(): ?string
 
     public function isExtendProductProtectionEnabled(): bool
     {
-        return $this->scopeConfig->getValue(Extend::ENABLE_PRODUCT_PROTECTION) === '1';
+        return $this->getScopedConfigValue(Extend::ENABLE_PRODUCT_PROTECTION) === '1';
     }
 
     public function isProductProtectionProductDisplayPageOfferEnabled(): bool
     {
-        return $this->scopeConfig->getValue(
+        return $this->getScopedConfigValue(
             Extend::ENABLE_PRODUCT_PROTECTION_PRODUCT_DISPLAY_PAGE_OFFER
         ) === '1';
     }
 
     public function isProductProtectionCartOfferEnabled(): bool
     {
-        return $this->scopeConfig->getValue(Extend::ENABLE_PRODUCT_PROTECTION_CART_OFFER) === '1';
+        return $this->getScopedConfigValue(Extend::ENABLE_PRODUCT_PROTECTION_CART_OFFER) === '1';
+    }
+    public function isProductProtectionMinicartOfferEnabled(): bool
+    {
+        return $this->getScopedConfigValue(Extend::ENABLE_PRODUCT_PROTECTION_MINICART_OFFER) ===
+            '1';
     }
 
     public function isProductProtectionPostPurchaseLeadModalOfferEnabled(): bool
     {
-        return $this->scopeConfig->getValue(
+        return $this->getScopedConfigValue(
             Extend::ENABLE_PRODUCT_PROTECTION_POST_PURCHASE_LEAD_MODAL_OFFER
         ) === '1';
     }
 
     public function isProductProtectionProductCatalogPageModalOfferEnabled(): bool
     {
-        return $this->scopeConfig->getValue(
+        return $this->getScopedConfigValue(
             Extend::ENABLE_PRODUCT_PROTECTION_PRODUCT_CATALOG_PAGE_MODAL_OFFER
         ) === '1';
     }
 
+    /**
+     * Get Scoped Config Value
+     *
+     * @param string $configPath
+     * @return string
+     */
+    private function getScopedConfigValue(string $configPath): string
+    {
+        $scopeCode = $this->storeManager->getStore()->getCode();
+        $scopeType = ScopeInterface::SCOPE_STORES;
+        return $this->scopeConfig->getValue($configPath, $scopeType, $scopeCode);
```

Comment:
question for the audience: do you know if i need to be more careful here about the case where there is no `store` returned by the `storeManager`? i assume in the case where `$scopeCode` is undefined, `getValue` will just look up the value for the website/default scope, traversing up the scopes, but i assume calling `getStore()->getCode()` where `getStore()` returns `undefined` is probably not good

Reply:
Should be wesbite/default by default

Reply:
I think it should be ok, when I stepped through it and that value was undefined it would grab the current store off the scope that it was called from